scripts/core/adapter.yml

```diff
@@ -177,6 +177,50 @@ returns:
     - $X_RESULT_ERROR_OUT_OF_RESOURCES
     - $X_RESULT_ERROR_OUT_OF_HOST_MEMORY
 --- #--------------------------------------------------------------------------
+type: fptr_typedef
+desc: "Callback function that processes adapter log messages."
+name: $x_logger_callback_t
+return: void
+params:
+  - type: $x_adapter_handle_t
+    name: hAdapter
+    desc: "[in] handle of the adapter"
+  - type: const char*
+    name: message
+    desc: "[in] the log message set by the unified runtime adapter."
+  - type: $x_log_level_t
+    name: logLevel
+    desc: "[in] The log message type"
```

Comment:
I feel like the log level should be set once as an argument to `urAdapterSetLoggingCallback`. This would mirror how it gets set by environment variable i.e. once per process. The main benefit of setting the log level once is to allow the logging framework to avoid paying the overhead of processing log messages which are going to be ignored.

Reply:
That makes sense. I added the extra parameter in the latest commit :+1: